DatadogRUM/Sources/Integrations/TelemetryReceiver.swift

```diff
@@ -255,11 +259,18 @@ internal final class TelemetryReceiver: FeatureMessageReceiver {
             let sessionIDOverride: String? = attributes.removeValue(forKey: SDKMetricFields.sessionIDOverrideKey)?.dd.decode()
             let sessionID = sessionIDOverride ?? rum?.sessionID
 
+            // Calculates the composition of sample rates. The metric can have up to 3 layers of sampling.
+            var effectiveSampleRate = metric.sampleRate.composed(with: self.sampler.samplingRate)
+            if let headSampleRate = metric.attributes[MethodCalledMetric.headSampleRate] as? SampleRate {
+                effectiveSampleRate = effectiveSampleRate.composed(with: headSampleRate)
+            }
```

Comment:
suggestion/ This is generic flow for any type of metric, but here we use `MethodCalledMetric.headSampleRate` which is specific to "Method Called" metric. It could make sense to move `.headSampleRate` to more generic [`SDKMetricFields`](https://github.com/DataDog/dd-sdk-ios/blob/a4c6a1cb49a22eec9f2d5dada8be160082533d35/DatadogInternal/Sources/SDKMetrics/SDKMetricFields.swift#L10) namespace, WDYT?


Also, the `.headSampleRate` is never removed from `attributes`, which means it is uploaded with the metric log, no? I think we should remove it here, similar to how we handle `SDKMetricFields.sessionIDOverrideKey` above.

Reply:
Makes sense. It is generic and has an appropriate enum. Also removed it from being uploaded. ty